pyfarm/agent/entrypoints/main.py

```diff
@@ -128,15 +128,23 @@ def __init__(self):
             help="The password required to access manipulate the agent "
                  "using REST. [default: %(default)s]")
         global_network.add_argument(
-            "--systemid", config="agent_systemid", type=system_identifier,
-            default=config["agent_systemid"],
-            help="The system identification value.  This is used to help "
-                 "identify the system itself to the master when the agent "
-                 "connects. [default: %(default)s]")
-        global_network.add_argument(
-            "--systemid-cache", config="agent_systemid_cache",
-            help="The location to cache the value for --systemid. "
+            "--agent-id", config="agent_id", type=uuid_type,
+            default=AgentUUID.load(),
+            help="The UUID used to identify this agent to the master.  By "
+                 "default the agent will attempt to load a cached value "
+                 "however a specific UUID could be provided with this flag. "
                  "[default: %(default)s]")
+        global_network.add_argument(
+            "--agent-id-cache", config="agent_id_cache",
```

Comment:
I think we should rename this to agent-id-store or agent-id-file. A cache is a performance optimisation. It stores data closer to where it will likely be needed, and it usually can be deleted/cleared/flushed/whatever without breaking anything.

The uuid.dat file OTOH has nothing to do with performance, and deleting it will break things.